packages/renderer/src/lib/troubleshooting/TroubleshootingGatherLogs.svelte

```diff
@@ -4,14 +4,17 @@ import Fa from 'svelte-fa';
 
 import Button from '/@/lib/ui/Button.svelte';
 
+import { Uri } from '../uri/Uri';
+
 let logs: string[] = [];
 
 // Save files as a zip file (we first ask the user for the dialog, and then save the files to the filepath)
 async function saveLogsAsZip() {
   const defaultUri = await window.troubleshootingGenerateLogFileUri('podman-desktop', 'zip');
   const filePath = await window.saveDialog({ title: 'Save Logs as .zip', defaultUri });
   if (filePath) {
-    logs = await window.troubleshootingSaveLogs(filePath);
+    const filePathUri = Uri.revive(filePath);
```

Comment:
Is it possible to have revive being done in the exposure of troubleshootingSaveLogs so we don't need that after each call ? Maybe it's not possible, just asking

Reply:
I tried to put the Uri class in the preload without any success. It still takes the private props as keys.
But if i create a revive function that just returns an object like 
```
  static revive(serialized: APIUri): APIUri {
    if (serialized instanceof Uri) {
      return serialized;
    }
    const serializedProps: Map<string, string> = Object.entries(serialized)
      .map(([key, value]) => [key.startsWith('_') ? key.substring(1) : key, value])
      .reduce((map, [key, value]) => {
        map.set(key, value);
        return map;
      }, new Map());
    return {
      scheme: serializedProps.get('scheme') ?? '',
      authority: serializedProps.get('authority') ?? '',
      path: serializedProps.get('path') ?? '',
      query: serializedProps.get('query') ?? '',
      fragment: serializedProps.get('fragment') ?? '',
    } as APIUri;
  }
```
it works fine. Do i do this?

Reply:
no I think it won't fit well, keep it as you have